pandas/core/reshape/concat.py

```diff
@@ -37,7 +37,7 @@ def concat(
     verify_integrity: bool = False,
     sort: bool = False,
     copy: bool = True,
-):
+) -> Union["DataFrame", "Series"]:
```

Comment:
I assume FrameOrSeries doesn't apply here?

Reply:
No, the problem is the return type doesn't always follow a parameter type. Specifically if the input are  series, and axis=1, the output is a DataFrame. We need Literals to do it better.